src/components/TagPicker/index.tsx

```diff
@@ -10,22 +10,64 @@ import * as OptionsListUtils from '@libs/OptionsListUtils';
 import * as PolicyUtils from '@libs/PolicyUtils';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
-import {defaultProps, propTypes} from './tagPickerPropTypes';
+import type {PolicyTag, PolicyTagList, PolicyTags, RecentlyUsedTags} from '@src/types/onyx';
 
-function TagPicker({selectedTag, tag, tagIndex, policyTags, policyRecentlyUsedTags, shouldShowDisabledAndSelectedOption, insets, onSubmit}) {
+type SelectedTagOption = {
+    name: string;
+    enabled: boolean;
+    accountID: number | null;
+};
+
+type TagPickerOnyxProps = {
+    /** Collection of tags attached to a policy */
+    policyTags: OnyxEntry<PolicyTagList>;
+
+    /** List of recently used tags */
+    policyRecentlyUsedTags: OnyxEntry<RecentlyUsedTags>;
+};
+
+type TagPickerProps = TagPickerOnyxProps & {
+    /** The policyID we are getting tags for */
+    // It's used in withOnyx HOC.
+    // eslint-disable-next-line react/no-unused-prop-types
+    policyID: string;
+
+    /** The selected tag of the money request */
+    selectedTag: string;
+
+    /** The name of tag list we are getting tags for */
+    tag: string;
```

Comment:
```suggestion
    tagListName: string;
```

Otherwise it's super confusing

Reply:
@flodnv done 😄